iohk-monitoring/src/Cardano/BM/Data/MonitoringEval.lhs

```diff
@@ -156,6 +184,57 @@ parseExpr = do
 
 \end{code}
 
+\label{code:parseAction}
+An action is enclosed in parentheses.
+\begin{code}
+parseAction :: P.Parser MEvAction
+parseAction =
+        (nextIsChar 'C' >> parseActionCreateMessage)
+    <|> (nextIsChar 'S' >> parseActionSetMinSeverity)
+    <|> (nextIsChar 'A' >> parseActionAlterSeverity)
+
+parseActionCreateMessage :: P.Parser MEvAction
+parseActionCreateMessage = do
+    void $ P.string "CreateMessage"
+    P.skipSpace
+    sev <- parsePureSeverity
+    P.skipSpace
+    void $ P.char '\"'
+    alertMessage <- P.takeWhile1 (/='\"')
+    void $ P.char '\"'
+    return $ CreateMessage sev alertMessage
+
+parseActionSetMinSeverity :: P.Parser MEvAction
+parseActionSetMinSeverity = do
+    void $ P.string "SetGlobalMinimalSeverity"
+    P.skipSpace
+    sev <- parsePureSeverity
+    return $ SetGlobalMinimalSeverity sev
+
+parseActionAlterSeverity :: P.Parser MEvAction
+parseActionAlterSeverity = do
+    void $ P.string "AlterSeverity"
+    P.skipSpace
+    void $ P.char '\"'
+    loggerName <- P.takeWhile1 (/='\"')
+    void $ P.char '\"'
+    P.skipSpace
+    sev <- parsePureSeverity
+    return $ AlterSeverity loggerName sev
+
+parsePureSeverity :: P.Parser Severity
+parsePureSeverity =
```

Comment:
I am sure there is already some parser for severity. But this can be changed later.